exec-service-wasmer/src/wasmer_instance.rs

```diff
@@ -10,6 +10,7 @@ use wasmer_engine_universal::Universal;
 pub struct WasmerInstance {
     pub executor_data: Rc<WasmerExecutorData>,
     pub wasmer_instance: wasmer::Instance,
+    memory_name: String,
```

Comment:
Why is the memory name necessary?

Reply:
It is kept a a key, so I don't have to extract an unsafe pointer to it the webassembly. You access the memory via

```
    fn get_memory_ref(&self) -> &wasmer::Memory {
        self.wasmer_instance
            .exports
            .get_memory(&self.memory_name)
            .unwrap_or_else(|_| {
                panic!("memory name not found, should not happen since it was already checked")
            })
    }
```

The alternative is an unsafe pointer to it in the webassembly .... it might work too, but I need to see how to hide the unsafe-ness.

Reply:
We just need to be careful with warm instances, not to end up with an invalid pointer ....